src/main/java/wolox/reactortraining/beans/WebClientBean.java

```diff
@@ -0,0 +1,18 @@
+package wolox.reactortraining.beans;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Component
+public class WebClientBean {
+
+    @Value("${api.bots.uri}")
+    private String botsUri;
+
+    @Bean("bot-api")
+    public WebClient botApiClient() {
+        return WebClient.create(botsUri);
+    }
+}
```

Comment:
Why create a bean for the webClient?, can be simplified by injecting it into the service builder

Reply:
As commented in the PR description, I opted to use scope-base injection to try out the feature.